scripts/tank_cmd.py

```diff
@@ -1230,7 +1254,148 @@ def run_engine_cmd(log, pipeline_config_root, context_items, command, using_cwd,
+    for a in found_args:
+        cmd_line.remove(a)
+
+    credentials = []
+
+    # Create a list of tuples for each argument we found.
+    for arg in found_args:
+        pos = arg.find("=")
+        # Take everything after the -- up to but not including the =
+        arg_name = arg[2:pos]
+        arg_value = arg[pos + 1:]
+        credentials.append((arg_name, arg_value))
+
+    return credentials
+
+
+def _validate_only_once(args, arg):
+    """
+    Validates that an argument is only present once.
+
+    :param args: List of tuples of arguments that were extracted from the command line.
+    :param arg: Argument to validate
+
+    :raises InvalidCredentials: If an argument has been specified more than once,
+                                this exception is raised.
+    """
+    occurences = filter(lambda a: a[0] == arg, args)
+    if len(occurences) > 1:
+        raise InvalidCredentials("argument '%s' specified more than once." % arg)
+
+
+def _validate_args_cardinality(args, arg1, arg2):
+    """
+    Makes sure that there are no more and no less than 1 value for each argument.
+
+    :param args: List of argument tuples.
+    :param arg1: First argument to test for.
+    :param arg2: Second argument to test for.
+
+    :raises InvalidCredentials: When there is more or less than 1 value for
+                                a given argument, this exception is raised.
+    """
+    # Too few parameters, find out which one is missing, let the user know
+    # which is missing.
+    if len(args) < 2:
+        raise InvalidCredentials("missing argument '%s'" % (arg1 if args[0][0] == arg2 else arg2))
+
+    # Make sure each arguments are not specified more than once.
+    _validate_only_once(args, arg1)
+    _validate_only_once(args, arg2)
+
+
+def _read_credentials_from_file(auth_path):
+    """
+    Reads the credentials from a file and returns two tuples: one for
+    login/password values and the other for script/key values. Any other values
+    are ignored.
+
+    :param auth_path: Path to a file with credentials.
+
+    :returns: A tuple of list of key, value pairs for values parsed. For example,
+       ([("login", "user"), ("password", "12345")],
+        [("script", "name"), ("key", "12345")])
+       The first tuple is concerned with human user based authentication, the second
+       is concerned with script based authentication.
+
+    :raises InvalidCredentials: If the file doesn't exist, this exception is raised.
+    """
+    if not os.path.exists(auth_path):
+        raise InvalidCredentials("credentials file does not exist.")
+    # Read the dictionary from file
+    with open(auth_path) as auth_file:
+        file_data = yaml.load(auth_file)
+
+    # We've already fixed the problem of parsing a bunch of values on the
+    # command line that may be mixed, so reshuffle the information that way
+    # and read from credentials
+    fake_cmd_line = ["--%s=%s" % (arg, value) for arg, value in file_data.iteritems()]
```

Comment:
This doesn't seem right to me! But maybe it's less of an issue if we only allow for script based logins?